rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake

```diff
@@ -15,111 +15,52 @@
-  elseif(_extension STREQUAL ".srv")
-    set(_allowed_parent_folders "srv" "action")
-    if(NOT _parent_folder IN_LIST _allowed_parent_folders)
-      message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
-    endif()
-    list(APPEND _generated_files "${_output_path}/${_parent_folder}/${_header_name}__rosidl_typesupport_opensplice_c.h")
-    list(APPEND _generated_files "${_output_path}/${_parent_folder}/dds_opensplice_c/${_header_name}__type_support_c.cpp")
-
-    foreach(_suffix "_Request" "_Response")
-      list(APPEND _generated_external_files
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_.h"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_.cpp"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_Dcps.h"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_Dcps.cpp"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_Dcps_impl.h"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_Dcps_impl.cpp"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_SplDcps.h"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/Sample_${_msg_name}${_suffix}_SplDcps.cpp"
-        "${_dds_output_path}/${_parent_folder}/dds_opensplice/ccpp_Sample_${_msg_name}${_suffix}_.h")
-    endforeach()
-  else()
-    message(FATAL_ERROR "Interface file with unknown extension: ${_idl_file}")
-  endif()
+  get_filename_component(_idl_name "${_abs_idl_file}" NAME_WE)
+  string_camel_case_to_lower_case_underscore("${_idl_name}" _header_name)
+  list(APPEND _dds_idl_files
+    "${_dds_idl_base_path}/${PROJECT_NAME}/${_parent_folder}/dds_opensplice/${_idl_name}_.idl")
+  list(APPEND _generated_files "${_output_path}/${_parent_folder}/${_header_name}__rosidl_typesupport_opensplice_c.h")
+  list(APPEND _generated_files "${_output_path}/${_parent_folder}/dds_opensplice_c/${_header_name}__type_support_c.cpp")
+  list(APPEND _generated_external_files
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_.h"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_.cpp"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_Dcps.h"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_Dcps.cpp"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_Dcps_impl.h"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_Dcps_impl.cpp"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_SplDcps.h"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/${_idl_name}_SplDcps.cpp"
+    "${_dds_output_path}/${_parent_folder}/dds_opensplice/ccpp_${_idl_name}_.h")
```

Comment:
Since these are generated by `rosidl_typesupport_opensplice_cpp`, can these names be gotten from that package? Or better yet, can `..._cpp` generate a library that `..._c` links against instead of adding them to it's own library?

Reply:
Maybe possible. But since this is the way it is on `master` I am not planning to change it within this PR.